src/EnergyPlus/HVACDXHeatPumpSystem.cc

```diff
@@ -726,52 +728,58 @@ namespace HVACDXHeatPumpSystem {
-                                Par(5) = double(FanOpMode);
-                                SolveRoot(Acc, MaxIte, SolFla, PartLoadFrac, DXHeatingCoilResidual, 0.0, 1.0, Par);
-                                if (SolFla == -1) {
-                                    if (!WarmupFlag) {
-                                        if (DXHeatPumpSystem(DXSystemNum).DXCoilSensPLRIter < 1) {
-                                            ++DXHeatPumpSystem(DXSystemNum).DXCoilSensPLRIter;
-                                            ShowWarningError(DXHeatPumpSystem(DXSystemNum).DXHeatPumpSystemType +
-                                                             " - Iteration limit exceeded calculating DX unit sensible part-load ratio for unit = " +
-                                                             DXHeatPumpSystem(DXSystemNum).Name);
-                                            ShowContinueError("Estimated part-load ratio  = " + RoundSigDigits((ReqOutput / FullOutput), 3));
-                                            ShowContinueError("Calculated part-load ratio = " + RoundSigDigits(PartLoadFrac, 3));
-                                            ShowContinueErrorTimeStamp(
-                                                "The calculated part-load ratio will be used and the simulation continues. Occurrence info:");
-                                        } else {
-                                            ShowRecurringWarningErrorAtEnd(
-                                                DXHeatPumpSystem(DXSystemNum).DXHeatPumpSystemType + " \"" + DXHeatPumpSystem(DXSystemNum).Name +
-                                                    "\" - Iteration limit exceeded calculating sensible part-load ratio error continues. Sensible "
-                                                    "PLR statistics follow.",
-                                                DXHeatPumpSystem(DXSystemNum).DXCoilSensPLRIterIndex,
-                                                PartLoadFrac,
-                                                PartLoadFrac);
+                                if (DataGlobals::DoCoilDirectSolutions) {
+                                    PartLoadFrac = (DesOutTemp - Node(InletNode).Temp) / (TempOut1 - Node(InletNode).Temp);
+                                    SimDXCoil(
+                                        CompName, On, FirstHVACIteration, DXHeatPumpSystem(DXSystemNum).HeatPumpCoilIndex, FanOpMode, PartLoadFrac);
+                                } else {
+                                    Par(1) = double(DXHeatPumpSystem(DXSystemNum).HeatPumpCoilIndex);
+                                    Par(2) = DesOutTemp;
+                                    Par(3) = 1.0; // OnOffAirFlowFrac assume = 1.0 for continuous fan dx system
+                                    Par(5) = double(FanOpMode);
+                                    SolveRoot(Acc, MaxIte, SolFla, PartLoadFrac, DXHeatingCoilResidual, 0.0, 1.0, Par);
```

Comment:
This is just a thought... so don't put too much weight on this comment. Could this be done in RootSolver so that no code would have to change?

Reply:
@rraustad Thanks for a good suggestion. Right now, I am testing all possible coils to see if the direct solution approach can be applied or not. Half work has been through. When I complete all possible coils, I will think how to consolidate the changes. Some changes are beyond my original thought. For example, a linear relationship between system output and part load ratio was proposed. The recent change to handle variable speed coils expands the application with a linear relationship between design output temperature and part load ratio. Therefore, final outcome will be based on all changes.